app/scripts/services/secrets.js

```diff
@@ -0,0 +1,31 @@
+'use strict';
+
+angular.module("openshiftConsole")
+  .factory("SecretsService", function(){
+
+    var groupSecretsByType = function(secrets, addEmptyValue) {
```

Comment:
`addEmptyValue` seems strange. I'd prefer just to do something like

```js
secrets = [''].concat(secrets);
```

in the controller and never add an empty value in the service. The empty value is for the dropdown? You shouldn't change the service to support that.